aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -213,6 +213,40 @@ void TensorIterator::allocate_outputs() {
   }
 }
 
+#ifdef BUILD_NAMEDTENSOR
+void TensorIterator::propagate_names_to_outputs() {
+  NameVector names;
+
+  // build names
+  for (auto& op : operands_) {
+    if (!op.tensor.defined()) continue;
+    // don't include output tensors that are not also input tensors.
+    if (resize_outputs_ && op.is_output && !op.is_read_write) continue;
+    // perform name inference
+    if (!op.tensor.is_named()) {
+      continue;
+    }
+    auto tensor_names = *op.tensor.names();
+    if (names.empty()) {
+      names = tensor_names;
+    } else {
+      names = NameVector(unify_from_right(names, tensor_names).value());
+    }
+  }
+  if (names.empty()) {
+    return;
+  }
+
+  // propagate names
+  for (int i = 0; i < num_outputs_; i++) {
+    auto& op = operands_[i];
```

Comment:
Do you happen to know why `Tensor output(int arg=0)` doesn't return a `Tensor&` ? Seems like it would be nice to be able to use that here.